### ISSUE TITLE
Implement New Line (O/o - "open") in Vi-mode

### ISSUE THREAD

**Issue by joallard** [fish-shell/fish-shell#7442]
One of my commands in Vi is 'open': quite practical, it inserts a new line below or above the current cursor, and starts insert mode there.

This is an enhancement ticket to track this feature.

Quoting the Vim docs:

```
							*o*
o			Begin a new line below the cursor and insert text,
			repeat [count] times.
			[...]

							*O*
O			Begin a new line above the cursor and insert text,
			repeat [count] times.
			[...]
```

Current workaround: Alt+Enter. Not bad! Though it splits at current cursor position.

Equivalent commands: `o`: `A<Enter>`, `O`: `I<Enter><Up>`

I'm not much familiar with Fish source, does someone know what implementing this would entail? Unless I'm mistaken, it doesn't sound too complex.

[enhancement] [vi-mode]

**Comment by ridiculousfish** [comment]
The basic steps are:

1. Add a new readline command in [input_common.h](https://github.com/fish-shell/fish-shell/blob/master/src/input_common.h)
2. Implement it in [`handle_readline_command`](https://github.com/fish-shell/fish-shell/blob/master/src/reader.cpp#L2731)
3. Add it to default vi bindings in [`fish_vi_key_bindings`](https://github.com/fish-shell/fish-shell/blob/master/share/functions/fish_vi_key_bindings.fish)

**Comment by tamis-laan** [comment]
> The basic steps are:
> 
> 1. Add a new readline command in [input_common.h](https://github.com/fish-shell/fish-shell/blob/master/src/input_common.h)
> 2. Implement it in [`handle_readline_command`](https://github.com/fish-shell/fish-shell/blob/master/src/reader.cpp#L2731)
> 3. Add it to default vi bindings in [`fish_vi_key_bindings`](https://github.com/fish-shell/fish-shell/blob/master/share/functions/fish_vi_key_bindings.fish)

If I would like to implement `ci"` `ci(` etc, where should I specify this?

**Comment by ridiculousfish** [comment]
Take a look inside [fish_vi_key_bindings.fish](https://github.com/fish-shell/fish-shell/blob/master/share/functions/fish_vi_key_bindings.fish), you can see how some other vim bindings are implemented.

**Comment by reach-satori** [comment]
[0001-vi-mode-o-and-O-commands.patch.txt](https://github.com/fish-shell/fish-shell/files/5688870/0001-vi-mode-o-and-O-commands.patch.txt)
I didn't wanna do a whole pull request for this because it's indeed very simple, plus I'm new to contributing. But here's a patch file if anyone wants to do something with it. (with modified filename, since github doesn't like .patch ending files). 

All it took was repeating the beginning-of-line and end-of-line cases and adding a newline character afterwards.

**Comment by ridiculousfish** [comment]
Thank you, merged as 2d2efc8b2ece97384bf08fc55c0eb35eaa86fb5f. Though I accidentally gave credit in the commit to the bug filer joallard, instead of @reach-satori - sorry about that!
